lib/lambda_ethereum_consensus/p2p/gossip/attestation.ex

```diff
@@ -98,41 +101,45 @@ defmodule LambdaEthereumConsensus.P2P.Gossip.Attestation do
 
   @impl true
   def init(_init_arg) do
-    {:ok, %{attnets: %{}, attestations: %{}}}
+    {:ok, nil}
   end
 
   @impl true
-  def handle_call({:collect, subnet_id, attestation}, _from, state) do
-    attestations = Map.put(state.attestations, subnet_id, [attestation])
-    attnets = Map.put(state.attnets, subnet_id, attestation.data)
-    new_state = %{state | attnets: attnets, attestations: attestations}
-    get_topic_name(subnet_id) |> Libp2pPort.subscribe_to_topic(__MODULE__)
-    {:reply, :ok, new_state}
-  end
-
-  def handle_call({:stop_collecting, subnet_id}, _from, state) do
-    if Map.has_key?(state.attnets, subnet_id) do
-      {collected, atts} = Map.pop(state.attestations, subnet_id, [])
-      new_state = %{state | attnets: Map.delete(state.attnets, subnet_id), attestations: atts}
-      {:reply, {:ok, collected}, new_state}
-    else
-      {:reply, {:error, "subnet not joined"}, state}
+  def handle_call({:collect, subnet_id, attestation}, _from, _state) do
+    subnet_info = SubnetInfo.new_subnet_with_attestation(attestation)
+    persist_subnet_info(subnet_info, subnet_id)
```

Comment:
nit: Let's put the subnet_id first. As a convention, we usually have our calls be `store(key, value)` instead of `store(value, key)`.